src/main/java/org/candlepin/subscriptions/jmx/OptInJmxBean.java

```diff
@@ -82,4 +92,30 @@ public String createOrUpdateOptInConfig(String accountNumber, String orgId, bool
         return String.format(text, accountNumber, orgId, config.toString());
     }
 
+    @ManagedAttribute(description = "Count of how many orgs opted-in in the previous week.")
+    public int getLastWeekOptInCount() {
+        OffsetDateTime oneWeekAgo = OffsetDateTime.now().minusWeeks(1);
+        OffsetDateTime begin = clock.startOfWeek(oneWeekAgo);
+        OffsetDateTime end = clock.endOfWeek(oneWeekAgo);
+        return repo.getCountOfOptInsForDateRange(begin, end);
+    }
+
+    @ManagedAttribute(description = "Count of how many orgs opted-in in the current week.")
+    public int getCurrentWeekOptInCount() {
+        OffsetDateTime now = OffsetDateTime.now();
+        OffsetDateTime begin = clock.startOfWeek(now);
+        OffsetDateTime end = clock.endOfWeek(now);
+        return repo.getCountOfOptInsForDateRange(begin, end);
+    }
+
+    @ManagedOperation(description = "Fetch the number of orgs opted-in in a given week.")
+    @ManagedOperationParameter(name = "weekOf", description = "Date in the week to query; YYYY-MM-DD format")
+    public int getOptInCountForWeekOf(String weekOf) throws ParseException {
+        OffsetDateTime dateInWeek = OffsetDateTime
+            .ofInstant(new SimpleDateFormat("yyyy-MM-dd").parse(weekOf).toInstant(), ZoneOffset.UTC);
+        OffsetDateTime begin = clock.startOfWeek(dateInWeek);
+        OffsetDateTime end = clock.endOfWeek(dateInWeek);
+        return repo.getCountOfOptInsForDateRange(begin, end);
+    }
+
```

Comment:
Let's DRY.  I'm writing this off the top of my head, so I'm not going to put it up as a GitHub suggestion, but I think you can just copy-paste it and clean up any erroneous syntax.

```java
@ManagedAttribute(description = "Count of how many orgs opted-in in the previous week.")
public int getLastWeekOptInCount() {
    OffsetDateTime weekDate = OffsetDateTime.now().minusWeeks(1);
    return weekCount(weekDate);
}
   
@ManagedAttribute(description = "Count of how many orgs opted-in in the current week.")
public int getCurrentWeekOptInCount() {
    OffsetDateTime weekDate = OffsetDateTime.now();
    return weekCount(weekDate);
}

@ManagedOperation(description = "Fetch the number of orgs opted-in in a given week.")
@ManagedOperationParameter(name = "weekOf", description = "Date in the week to query; YYYY-MM-DD format")
public int getOptInCountForWeekOf(String weekOf) throws ParseException {
    OffsetDateTime weekDate = OffsetDateTime
        .ofInstant(new SimpleDateFormat("yyyy-MM-dd").parse(weekOf).toInstant(), ZoneOffset.UTC);
    return weekCount(weekDate);
}

protected int weekCount(OffsetDateTime weekDate) {
    OffsetDateTime begin = clock.startOfWeek(weekDate);
    OffsetDateTime end = clock.endOfWeek(weekDate);
    return repo.getCountOfOptInsForDateRange(begin, end);
}
```